electrumsv/tests/test_wallet.py

```diff
@@ -550,64 +621,249 @@ def test_legacy_wallet_loading(mock_wallet_app_state, storage_info: WalletStorag
-        check_legacy_parent_of_multisig_wallet(wallet, password)
+        check_legacy_parent_of_multisig_wallet(wallet, new_password)
     elif "hardware" == expected_type:
         check_legacy_parent_of_hardware_wallet(wallet)
     elif "blank" == expected_type:
         check_parent_of_blank_wallet(wallet)
     else:
         raise Exception(f"unrecognised wallet file {wallet_filename}")
 
-    check_specific_wallets(wallet, password, storage_info)
+    check_specific_wallets(wallet, new_password, storage_info)
 
-    if expected_network in { "testnet", "regtest" }:
-        Net.set_to(SVMainnet)
+    ## Cached data AFTER the password update
+    password_token_after_cached = wallet.get_storage().get('password-token')
+    assert pw_decode(password_token_after_cached, new_password) is not None
+    assert pw_decode(password_token_before_cached, initial_password) is not None
+    test_state.count_of_password_token_reencryptions_cache += 1
 
+    if "imported" == expected_type and "privkey" in wallet_filename:
+        keyinstance_private_keys_cached_after = get_cached_imported_privkeys(new_password)
+        assert keyinstance_private_keys_cached_after == keyinstance_private_keys_cached
+        test_state.count_of_keyinstance_privkey_reencryptions_cache += \
+            len(keyinstance_private_keys_cached_after)
+
+    encrypted_api_keys_cached_after = get_cached_server_api_keys(new_password)
+    assert encrypted_api_keys_cached_after == encrypted_api_keys_cached
+    test_state.count_of_server_api_key_reencryptions_cache += \
+        len(encrypted_api_keys_cached)
+
+    master_key_xprvs_cached_after, master_key_seeds_cached_after, \
+        master_key_passphrases_cached_after = get_cached_master_keys_encrypted_data(new_password)
+    assert master_key_xprvs_cached_after == master_key_xprvs_cached
+    assert master_key_seeds_cached_after == master_key_seeds_cached
+    assert master_key_passphrases_cached_after == master_key_passphrases_cached
+    test_state.count_of_masterkeys_table_xprv_reencryptions_cache += \
+        len(master_key_xprvs_cached_after)
+    test_state.count_of_masterkeys_table_seed_reencryptions_cache += \
+        len(master_key_seeds_cached_after)
+    test_state.count_of_masterkeys_table_passphrase_reencryptions_cache += \
+        len(master_key_passphrases_cached_after)
+
+    ## Database level data AFTER the password update
+    password_token_after = ""
+    for wallet_data_row in db_functions.read_wallet_datas(wallet.get_db_context()):
+        if wallet_data_row.key == "password-token":
+            password_token_after = wallet_data_row.value
+
+    assert pw_decode(password_token_after, new_password) is not None
+    assert pw_decode(password_token_before, initial_password) is not None
+    assert pw_decode(password_token_after, new_password) == \
+           pw_decode(password_token_after_cached, new_password)
+    test_state.count_of_password_token_reencryptions += 1
```

Comment:
What is the value in this? It counts every wallet? It has no meaning whatsoever?